From 102c600bbb64b23679a3b143cdf9815733bdcb55 Mon Sep 17 00:00:00 2001
From: jfarrell <jfarrell@apache.org>
Date: Thu, 15 Aug 2013 21:20:19 -0400
Subject: [PATCH] Thrift-2124: .o, .so, .la, .deps, .libs, gen-* files left
 tutorials, test and lib/cpp when making DIST Client: build Patch: jfarrell

Fixing all the missing files and dist build issues that have been ignored.
---
 Makefile.am                     | 25 +++++++++++++------------
 compiler/cpp/Makefile.am        |  3 ++-
 compiler/cpp/test_parser.sh     | 31 -------------------------------
 lib/php/Makefile.am             |  7 ++++++-
 test/Makefile.am                |  1 +
 tutorial/Makefile.am            | 28 +++++++++++++++-------------
 tutorial/go/Makefile.am         |  5 ++++-
 tutorial/java/Makefile.am       |  6 +++++-
 tutorial/js/Makefile.am         |  5 +++++
 tutorial/py.twisted/Makefile.am |  3 ++-
 10 files changed, 53 insertions(+), 61 deletions(-)
 delete mode 100644 compiler/cpp/test_parser.sh

diff --git a/Makefile.am b/Makefile.am
index 18749c3..af25ad3 100755
--- a/Makefile.am
+++ b/Makefile.am
@@ -25,22 +25,23 @@ if WITH_TESTS
 SUBDIRS += test
 endif
 
+SUBDIRS += tutorial
+
 dist-hook:
-	find $(distdir) -type f \( -iname ".deps" -or -iname ".libs" -or -iname ".gitignore" \
-		-or -iname ".DS_Store" -or -iname "._*" \) | xargs rm -rf
+	find $(distdir) -type f \( -iname ".DS_Store" -or -iname "._*" -or -iname ".gitignore" \) | xargs rm -rf
+	find $(distdir) -type d \( -iname ".deps" -or -iname ".libs" \) | xargs rm -rf
 	find $(distdir) -type d \( -iname ".svn" -or -iname ".git" \) | xargs rm -rf
 
 print-version:
 	@echo $(VERSION)
 
 EXTRA_DIST = \
-			.travis.yml \
-			contrib \
-			debian \
-			doc \
-			doap.rdf \
-			sonar-project.properties \
-			tutorial \
-			LICENSE \
-			CHANGES \
-			NOTICE
+	.travis.yml \
+	contrib \
+	debian \
+	doc \
+	doap.rdf \
+	sonar-project.properties \
+	LICENSE \
+	CHANGES \
+	NOTICE
diff --git a/compiler/cpp/Makefile.am b/compiler/cpp/Makefile.am
index 2c3e5dc..13fed07 100644
--- a/compiler/cpp/Makefile.am
+++ b/compiler/cpp/Makefile.am
@@ -101,7 +101,8 @@ libparse_a_SOURCES = src/thrifty.yy \
 WINDOWS_DIST = \
              compiler.sln \
              compiler.vcxproj \
-             compiler.vcxproj.filters
+             compiler.vcxproj.filters \
+             README_Windows.txt
 
 EXTRA_DIST = \
              $(WINDOWS_DIST)
diff --git a/compiler/cpp/test_parser.sh b/compiler/cpp/test_parser.sh
deleted file mode 100644
index 066e09a..0000000
--- a/compiler/cpp/test_parser.sh
+++ /dev/null
@@ -1,31 +0,0 @@
-#!/bin/bash
-
-# Tests the parser, independently of whether any generators
-# are correct or useful.
-# Currently only tests that valid .thrift files parse cleanly.
-# Doesn't test that correct information is extracted from them.
-
-shopt -s extglob
-
-MY_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-ROOT_DIR=`cd $MY_DIR/../../ && pwd`
-TEST_THRIFT_DIR=${ROOT_DIR}/test
-THRIFT_FILES=`find ${TEST_THRIFT_DIR} -type f -name *.thrift ! -name BrokenConstants.thrift`
-
-OUTPUT_DIR=`mktemp -d -t test_thrift_parser.XXXXX`
-
-PASS=0
-FAIL=0
-for f in ${THRIFT_FILES};
-do
-  echo "Parsing ${f}"
-  ${MY_DIR}/thrift -o ${OUTPUT_DIR} -nowarn --allow-64bit-consts --gen cpp ${f}
-  EXIT_CODE=$?
-  if [ ${EXIT_CODE} -eq 0 ]; then
-    let PASS=PASS+1
-  else
-    let FAIL=FAIL+1
-  fi
-done
-echo
-echo "${PASS} files parsed correctly. ${FAIL} files failed to parse."
diff --git a/lib/php/Makefile.am b/lib/php/Makefile.am
index 011740f..8e5e094 100755
--- a/lib/php/Makefile.am
+++ b/lib/php/Makefile.am
@@ -111,8 +111,13 @@ phptype_DATA = \
   lib/Thrift/Type/TType.php
 
 EXTRA_DIST = \
-	src \
 	lib \
+  src/autoload.php \
+  src/ext/thrift_protocol/config.w32 \
+  src/ext/thrift_protocol/php_thrift_protocol.cpp \ 
+  src/ext/thrift_protocol/php_thrift_protocol.h \
+  src/Thrift.php \
+  src/TStringUtils.php \
 	thrift_protocol.ini \
 	README.apache
 
diff --git a/test/Makefile.am b/test/Makefile.am
index 96a8a53..175f477 100755
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -70,6 +70,7 @@ EXTRA_DIST = \
 	DebugProtoTest.thrift \
 	DenseLinkingTest.thrift \
 	DocTest.thrift \
+	Include.thrift \
 	JavaBeansTest.thrift \
 	ManyOptionals.thrift \
 	ManyTypedefs.thrift \
diff --git a/tutorial/Makefile.am b/tutorial/Makefile.am
index 72901a5..5f65802 100755
--- a/tutorial/Makefile.am
+++ b/tutorial/Makefile.am
@@ -32,14 +32,6 @@ SUBDIRS += java
 SUBDIRS += js
 endif
 
-if WITH_PERL
-#SUBDIRS += perl
-endif
-
-if WITH_PHP
-#SUBDIRS += php
-endif
-
 if WITH_PYTHON
 SUBDIRS += py
 SUBDIRS += py.twisted
@@ -50,10 +42,6 @@ if WITH_RUBY
 SUBDIRS += rb
 endif
 
-if WITH_HASKELL
-#SUBDIRS += hs
-endif
-
 if WITH_GO
 SUBDIRS += go
 endif
@@ -63,5 +51,19 @@ endif
 #
 all-local:
 	$(top_builddir)/compiler/cpp/thrift --gen html -r $(top_srcdir)/tutorial/tutorial.thrift
-
 endif
+
+# Any folders or files not listed above being added to SUBDIR need to be placed here in 
+# EXTRA_DIST to be included in the release 
+EXTRA_DIST = \
+	csharp \
+	d \
+	delphi \
+	erl \
+	gen-html \
+	hs \
+	ocaml \
+	perl \
+	php \
+	shared.thrift \
+	tutorial.thrift
diff --git a/tutorial/go/Makefile.am b/tutorial/go/Makefile.am
index 5df065a..53c3980 100644
--- a/tutorial/go/Makefile.am
+++ b/tutorial/go/Makefile.am
@@ -55,4 +55,7 @@ EXTRA_DIST = \
 	src/client.go \
 	src/handler.go \
 	src/server.go \
-	src/main.go
+	src/main.go \
+	server.crt \
+	server.key
+
diff --git a/tutorial/java/Makefile.am b/tutorial/java/Makefile.am
index 3b2174a..aa8b80c 100755
--- a/tutorial/java/Makefile.am
+++ b/tutorial/java/Makefile.am
@@ -37,4 +37,8 @@ tutorialserver: all
 	$(ANT) $(ANT_FLAGS) tutorialserver
 
 tutorialclient: all
-	$(ANT) $(ANT_FLAGS) tutorialclient
\ No newline at end of file
+	$(ANT) $(ANT_FLAGS) tutorialclient
+
+EXTRA_DIST = \
+	build.xml \
+	src
diff --git a/tutorial/js/Makefile.am b/tutorial/js/Makefile.am
index c174500..3fe0888 100755
--- a/tutorial/js/Makefile.am
+++ b/tutorial/js/Makefile.am
@@ -32,3 +32,8 @@ check-local: all
 
 tutorialserver: all
 	$(ANT) $(ANT_FLAGS) tutorialserver
+
+EXTRA_DIST = \
+	build.xml \
+	src \
+	tutorial.html
diff --git a/tutorial/py.twisted/Makefile.am b/tutorial/py.twisted/Makefile.am
index 3a553eb..c6cbd45 100755
--- a/tutorial/py.twisted/Makefile.am
+++ b/tutorial/py.twisted/Makefile.am
@@ -34,5 +34,6 @@ clean-local:
 	$(RM) -r gen-*
 
 EXTRA_DIST = \
+	PythonClient.py \
 	PythonServer.py \
-	PythonClient.py
+	PythonServer.tac
-- 
1.8.3.1

